src/theme-context.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { createContext, useContext, useState } from 'react';
+
+interface ThemeContextType {
+	isDark: boolean;
+	toggleMode: () => void;
+}
+
+interface ThemeProviderProps {
+	children: React.ReactNode;
+}
+
+const ThemeContext = createContext<ThemeContextType | undefined>(undefined);
+
+export const ThemeProvider: React.FC<ThemeProviderProps> = ({ children }) => {
```

Comment:
This is not how we should implement themes, we should use the styled components theming approach
https://styled-components.com/docs/advanced